app/models/recipe.rb

```diff
@@ -1,16 +1,69 @@
 require 'net/http'
 class Recipe < ActiveRecord::Base
     def self.find_in_api(calories, budget, time)
-        base_uri = 'https://spoonacular-recipe-food-nutrition-v1.p.rapidapi.com/recipes/mealplans/generate'
-        uri = URI(base_uri + '?timeFrame=week&targetCalories=' + calories.to_s)
+        base_uri = 'https://spoonacular-recipe-food-nutrition-v1.p.rapidapi.com'
+        uri = URI(base_uri + '/recipes/mealplans/generate' + '?timeFrame=week&targetCalories=' + calories.to_s)
         req = Net::HTTP::Get.new(uri)
         req["X-RapidAPI-Host"] = "spoonacular-recipe-food-nutrition-v1.p.rapidapi.com"
         req["X-RapidAPI-Key"] = ENV["SPOONACULAR_API_KEY"]
 
         res = Net::HTTP.start(uri.hostname, :use_ssl => true) {|http|
             http.request(req)
         }
-        #puts JSON.pretty_generate(JSON.parse(res.body))
-        return res
+        res_json_hash = JSON.parse(res.body)
```

Comment:
run-on code, might be best to break up each of the three different api calls into their own methods?